jwcrypto/jwt.py

```diff
@@ -225,6 +228,14 @@ def __init__(self, header=None, claims=None, jwt=None, key=None,
             self._check_claims = check_claims
 
         if claims is not None:
+            # Check for datetime objects
+            if 'exp' in claims and isinstance(claims['exp'], datetime):
+                # Check if timezone aware
+                claims['exp'] = self._check_and_convert_dt(claims['exp'], 'exp')
```

Comment:
I do not think we should modify the array being passed in.
We should instead make a copy in self.claims and modify them there